bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java

```diff
@@ -115,6 +171,11 @@ public void added(ItemChannelLink itemChannelLink) {
 
         @Override
         public void removed(ItemChannelLink itemChannelLink) {
+            if (itemRegistry.get(itemChannelLink.getItemName()) == null) {
+                // Don't inform about the unlink if the item does not exist.
+                // The handler is informed if the item has been removed.
+                return;
```

Comment:
@kaikreuzer I assume this here is problematic.

If a link is removed and the item does not exist, this lines assume that the unlink is already sent by the item removal listener.

See: https://github.com/eclipse/smarthome/blob/6a8cb49/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java#L307
If a link is removed the ChannelItemProvider removes the item (if it is owned by him), too.

If the item and link is removed before the listener functions are called, no unlink of the thing manager is called.
Here we have "zero or one" handler unlink notifiaction(s).

Do you think the "removal" could be send for both, if the item is removed and the link is removed?
Then we have "one or two" handler unlink notifiaction(s).

I assume all that event handling is async. and the only way to really "one" notification is if we cache all item names internally. But I don't think this is memory efficient.

WDYT?